spec/custom/index.html

```diff
@@ -584,167 +459,225 @@
-<dt>Output</dt>
-    <dd><var>CONSTRUCTOR</var>, the <a>custom element constructor</a></dd>
-</dl>
-<ol class='algorithm'>
-    <li>If <var>FUNCTION</var> is <strong>null</strong>:
-    <ol>
-        <li>Let <var>FUNCTION</var> be the result of calling <a href="https://tc39.github.io/ecma262/#sec-functionallocate"><code>FunctionAllocate</code></a> with <a href="https://html.spec.whatwg.org/multipage/dom.html#htmlelement"><code>HTMLElement</code></a> as the <em>functionPrototype</em> and <strong>true</strong> as <em>strict</em></li>
-        <li>Let <var>PROTOTYPE</var> be the result of calling <a href="https://tc39.github.io/ecma262/#sec-objectcreate">Object.create</a> with <a href="https://html.spec.whatwg.org/multipage/dom.html#htmlelement"><code>HTMLElement</code></a>'s <a href="https://heycam.github.io/webidl/#dfn-interface-prototype-object">interface prototype object</a> as only argument</li>
-        <li>Call <code><a href="https://tc39.github.io/ecma262/#sec-definepropertyorthrow">DefinePropertyOrThrow</a>(</code><var>PROTOTYPE</var><code>, "constructor", PropertyDescriptor{[[\Value]]: </code><var>FUNCTION</var><code>, [[\Writable]]: false, [[\Enumerable]]: false, [[\Configurable]]: false})</code></li>
-        <li>Call <code><a href="https://tc39.github.io/ecma262/#sec-definepropertyorthrow">DefinePropertyOrThrow</a>(</code><var>FUNCTION</var><code>, "prototype", PropertyDescriptor{[[\Value]]: </code><var>PROTOTYPE</var><code>, [[\Writable]]: false, [[\Enumerable]]: false, [[\Configurable]]: false})</code></li>
-    </ol></li>
-    <li>Otherwise:
-    <ol>
-        <li>Let <var>PROTOTYPE</var> be the result of <code><a href="https://tc39.github.io/ecma262/#sec-get-o-p">Get</a>(</code><var>FUNCTION</var><code>, "prototype")</code></li>
-    </ol></li>
+<ol>
+    <li>Let <var>prototype</var> be <var>definition</var>'s <a href="#dfn-element-definition-prototype">prototype</a>.</li>
 
-    <li>Let <var>NAME</var> be <var>EXTENDS</var></li>
-    <li>Let <var>ERROR</var> be the result of running the <a>element registration algorithm</a> with <var>DOCUMENT</var>, <var>TYPE</var>, <var>PROTOTYPE</var>, and <var>NAME</var> as arguments</li>
-    <li>If <var>ERROR</var> is <code>InvalidType</code>, <a href="https://heycam.github.io/webidl/#dfn-throw">throw</a> a <code>SyntaxError</code> and <strong>stop</strong>.</li>
-    <li>If <var>ERROR</var> is not <code>None</code>, <a href="https://heycam.github.io/webidl/#dfn-throw">throw</a> a <code>NotSupportedError</code> and <strong>stop</strong>.</li>
-    <li>Return result of running <a data-lt="custom element constructor generation algorithm">custom element constructor generation algorithm</a> with <var>PROTOTYPE</var>, <var>FUNCTION</var>, and <var>DOCUMENT</var> as arguments.</li>
-</ol>
-</div>
+    <li>Perform <var>element</var>.[[\SetPrototypeOf]](<var>prototype</var>). Rethrow any exceptions.</li>
```

Comment:
It's probably best if performed this inside the HTMLElement constructor so that in the case the constructor throws before finishing super call, we wouldn't leave the element in a weird state.